avm/src/daemon/daemon.rs

```diff
@@ -213,14 +225,75 @@ async fn get_v1_stats() -> VMStatsV1 {
 // returns cluster delta
 async fn post_v1_stats(cluster_id: &str, deploy_token: &str) -> String {
   let vm_stats = get_v1_stats().await;
-  let stats_body = json!({
+  let mut stats_body = json!({
     "deployToken": deploy_token,
     "vmStats": vm_stats,
     "clusterId": cluster_id,
   });
+  let secret_string = SECRET_STRING.get().unwrap();
+  if let Some(secret_string) = secret_string.as_ref() {
+    stats_body
+      .as_object_mut()
+      .unwrap()
+      .insert("secretString".to_string(), json!(secret_string));
+  }
   post_v1("stats", stats_body).await
 }
 
+async fn control_port(req: Request<Body>) -> Result<Response<Body>, Infallible> {
+  let secret_string = SECRET_STRING.get().unwrap();
+  if secret_string.is_some() && !req.headers().contains_key(secret_string.as_ref().unwrap()) {
+    // If this control port is guarded by a secret string, make sure there's a header with that
+    // secret as the key (we don't care about the value) and abort otherwise
+    Ok(Response::builder().status(500).body("fail".into()).unwrap())
+  } else if TcpStream::connect("127.0.0.1:443").is_err() {
+    // If the Alan HTTPS server has not yet started, mark as a failure
+    Ok(Response::builder().status(500).body("fail".into()).unwrap())
+  } else if Path::new("./Dockerfile").exists()
+    && Path::new("./app.tar.gz").exists()
+    && TcpStream::connect("127.0.0.1:8088").is_err()
+  {
+    // If this is an Anycloud deployment and the child process hasn't started, mark as a failure
+    // TODO: Any way to generalize this so we don't have special logic for Anycloud?
+    Ok(Response::builder().status(500).body("fail".into()).unwrap())
+  } else {
+    // Everything passed, send an ok
+    Ok(Response::builder().status(200).body("ok".into()).unwrap())
+  }
+}
+
+// Used by the `daemon` mode only
+pub async fn run_agz_b64(agz_b64: &str, priv_key_b64: Option<&str>, cert_b64: Option<&str>) {
```

Comment:
well we can just remove the comment and the `pub`? ;)

Reply:
Oh, right. Will do.